Mathlib/Order/Birkhoff.lean

```diff
@@ -89,18 +88,81 @@ variable [Finite α]
     exact supIrred_Iic _
 
 end LowerSet
+section PartialOrder
+variable [PartialOrder α] [Fintype α]
+
+open LowerSet
+
+open scoped Classical
+
+variable (α)
+
+/-- **Birkhoff's Embedding Theorem**. Any nonempty finite distributive lattice can be embedded
+into its lattice of sup-irreducible lower sets. -/
+def OrderEmbedding.supIrredLowerSet : α ↪o {s : LowerSet α // SupIrred s} where
+  toFun a := ⟨LowerSet.Iic a, by simp only [supIrred_Iic]⟩
+  inj' _ := by simp_all only [Subtype.mk.injEq, LowerSet.Iic_inj, implies_true]
+  map_rel_iff' := by simp only [Function.Embedding.coeFn_mk, Subtype.mk_le_mk, Iic_le, mem_Iic_iff,
+    implies_true]
+
+variable {α}
+
+lemma OrderEmbedding.supIrredLowerSet_apply {a : α} {s : LowerSet α} (ha : LowerSet.Iic a = s) :
+    OrderEmbedding.supIrredLowerSet α a = s := by
+  unfold OrderEmbedding.supIrredLowerSet
+  simp_all only [supIrred_iff_of_finite, RelEmbedding.coe_mk, Embedding.coeFn_mk]
+
+-- Birkhoff's Embedding is actually surjective
```

Comment:
Make that comment a docstring or remove it

Reply:
Removed it